areas/sql/src/AzureMcp.Sql/Services/SqlService.cs

```diff
@@ -26,44 +26,21 @@ public class SqlService(ISubscriptionService subscriptionService, ITenantService
     {
         try
         {
-            var subscriptionResource = await _subscriptionService.GetSubscription(subscription, null, retryPolicy);
-
-            var resourceGroupResource = await subscriptionResource
-                .GetResourceGroupAsync(resourceGroup, cancellationToken);
-
-            var sqlServerResource = await resourceGroupResource.Value
-                .GetSqlServers()
-                .GetAsync(serverName);
-
-            var databaseResource = await sqlServerResource.Value
-                .GetSqlDatabases()
-                .GetAsync(databaseName);
-
-            var database = databaseResource.Value.Data;
+            var result = await ExecuteSingleResourceQueryAsync(
```

Comment:
If I remember correctly, ARG query has a hard limit on how many items it can return in one result. The old implementation uses GetAllAsync() which I suppose internally enumerates all the items before returning. I don't see the BaseAzureResourceService class handling paging for enumerating paged ARG results. Should we document this as a breaking change?

Reply:
Good point. After discussing with Xiang and following Jeffery's recommendation, I’ve set a default limit of 50 for the ARG query. I’ll update this method to support pagination once a conclusion is reached in https://github.com/modelcontextprotocol/modelcontextprotocol/discussions/799.

Reply:
Let's not hold on 799 for pagination.  I don't mind doing 50 for now, but the gate isn't 799, we can design it and implement it as we see fit - independent of what is in the mcp protocol.